dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResultHydrated.java

```diff
@@ -0,0 +1,183 @@
+                    return parentPath.startsWith(tempHref);
+                } else {
+                    // If it's a page, we check instead if it's the current URI
+                    return !isCodeLink() && navResult.getHref()
+                        .equalsIgnoreCase(reqURI);
+                }
+            }
+        }
+        return false;
+    }
+    @Override
+    public String getCodeLink() {
+        if (navResult.getCodeLink() != null && (navResult.getCodeLink()
+            .contains("$")
+                || navResult.getCodeLink()
+                    .contains("#"))) {
+            return UtilMethods.evaluateVelocity(navResult.getCodeLink(), context.getVelocityContext());
+        } else {
+            return navResult.getCodeLink();
+        }
+    }
+    @Override
+    public String getTitle() throws Exception {
+        return navResult.getTitle();
+    }
+
+    @Override
+    public String getHostId() {
+        return navResult.getHostId();
+    }
+
+
+    public String getHref() {
+        return navResult.getHref();
+    }
+
+    @Override
+    public boolean isShowOnMenu() {
+        return navResult.isShowOnMenu();
+    }
+
+    @Override
+    public boolean isCodeLink() {
+        return navResult.isCodeLink();
+    }
+
+    @Override
+    public int getOrder() {
+        return navResult.getOrder();
+    }
+
+
+    @Override
+    public boolean isFolder() {
+        return navResult.isFolder();
+    }
+
+    @Override
+    public List<? extends NavResult> getChildren() throws Exception {
+
+
+        List<NavResultHydrated> list = navResult.getChildren().stream().map(result -> new NavResultHydrated(result, this.context)).collect(Collectors.toList());
+        
+        
+        
+        if (Config.getBooleanProperty("ENABLE_NAV_PERMISSION_CHECK", false)) {
+            // now filtering permissions
+            List<NavResult> allow = new ArrayList<NavResult>(list.size());
+
+            HttpServletRequest req = (HttpServletRequest) context.getRequest();
+            User currentUser = WebAPILocator.getUserWebAPI()
+                .getLoggedInUser(req);
+            if (currentUser == null)
+                currentUser = APILocator.getUserAPI()
+                    .getAnonymousUser();
+            for (NavResult nv : list) {
+                try {
+                    if (APILocator.getPermissionAPI()
+                        .doesUserHavePermission(nv, PermissionAPI.PERMISSION_READ, currentUser)) {
+                        allow.add(nv);
+                    }
+                } catch (Exception ex) {
+                    Logger.error(this, ex.getMessage(), ex);
+                }
+            }
+            return allow;
+        }
+        return list;
+    }
+
+
+    public String getParentPath() throws DotDataException, DotSecurityException {
+        return navResult.getParentPath();
+    }
+
+
+
+    public NavResult getParent() throws DotDataException, DotSecurityException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1161)